.github/bin/common.sh

```diff
@@ -164,39 +164,91 @@ installYq() {
+                 )
+  CRDS_KINDS=($(kubectl get crds -o jsonpath="{.items[*].spec.names.plural}"))
+  KINDS=("${STANDARD_KINDS[@]}" "${CRDS_KINDS[@]}")
+
+  for kind in "${KINDS[@]}" ; do
+    dir="${ARTIFACTS_DIR}/resources/namespaced/${namespace}/${kind}"
+    mkdir -p $dir
+
+    names=$(kubectl get -n $namespace $kind --no-headers=true -o custom-columns=":metadata.name")
+    for name in $names ; do
+      name=${name//[:<>|*?]/_}
+      kubectl get -n $namespace $kind $name -o yaml > "${dir}/${name}.yaml"
+    done
+  done
+}
 
-  #determine the name of the devworkspace controller manager pod
-  local CONTROLLER_POD_NAME=$(oc get pods -n devworkspace-controller -l app.kubernetes.io/name=devworkspace-controller -o json | jq -r '.items[0].metadata.name')
-  local WEBHOOK_SVR_POD_NAME=$(oc get pods -n devworkspace-controller -l app.kubernetes.io/name=devworkspace-webhook-server -o json | jq -r '.items[0].metadata.name')
+collectClusterScopeK8sResources() {
+  declare -a KINDS=("crds"
+                    "clusterroles" "clusterrolebindings"
+                   )
+  for kind in "${KINDS[@]}" ; do
+    dir="${ARTIFACTS_DIR}/resources/cluster/${kind}"
+    mkdir -p $dir
 
-  # save the logs of all the containers in the DWO pod
-  for container in $(oc get pod -n devworkspace-controller ${CONTROLLER_POD_NAME} -o json | jq -r '.spec.containers[] | .name'); do
-    mkdir -p ${ARTIFACTS_DIR}/devworkspace-operator/${CONTROLLER_POD_NAME}
-    oc logs -n devworkspace-controller deployment/devworkspace-controller-manager -c ${container} > ${ARTIFACTS_DIR}/devworkspace-operator/${CONTROLLER_POD_NAME}/${container}.log
+    names=$(kubectl get -n $namespace $kind --no-headers=true -o custom-columns=":metadata.name")
+    for name in $names ; do
+      name=${name//[:<>|*?]/_}
+      kubectl get -n $namespace $kind $name -o yaml > "${dir}/${name}.yaml"
+    done
   done
+}
+
+collectPodsLogsForNamespace() {
+  namespace="$1"
+  if [[ -z $namespace ]]; then return; fi
 
-  for container in $(oc get pod -n devworkspace-controller ${WEBHOOK_SVR_POD_NAME} -o json | jq -r '.spec.containers[] | .name'); do
-    mkdir -p ${ARTIFACTS_DIR}/devworkspace-operator/${WEBHOOK_SVR_POD_NAME}
-    oc logs -n devworkspace-controller deployment/devworkspace-webhook-server -c ${container} > ${ARTIFACTS_DIR}/devworkspace-operator/${WEBHOOK_SVR_POD_NAME}/${container}.log
+  dir="${ARTIFACTS_DIR}/cluster/namespaces/${namespace}/logs"
+  mkdir -p $dir
+
+  pods=$(kubectl get -n $namespace pods --no-headers=true -o custom-columns=":metadata.name")
+  for pod in $pods ; do
+    containers=$(kubectl get -n $namespace pod $pod -o jsonpath="{.spec.containers[*].name}")
+    for container in $containers ; do
+      kubectl logs -n $namespace $pod -c $container > "${dir}/${pod}_${container}.log"
+    done
   done
 }
 
+collectEventsForNamespace() {
+  namespace="$1"
+  if [[ -z $namespace ]]; then return; fi
+
+  dir="${ARTIFACTS_DIR}/cluster/namespaces/${namespace}"
```

Comment:
`resources/namespaced` ?
pls move to a const to reuse.

Reply:
ok